File: CSharp/UnicodeFileToHtmlTextConverter/UnicodeFileToHtmlTextConverter.cs
using System.IO;

namespace TDDMicroExercises.UnicodeFileToHtmlTextConverter
{
    public class UnicodeFileToHtmlTextConverter
    {
        private string _fullFilenameWithPath;

        public UnicodeFileToHtmlTextConverter(string fullFilenameWithPath)
        {
            _fullFilenameWithPath = fullFilenameWithPath;
        }

        public string GetFilename()
        {
            return _fullFilenameWithPath;
        }

        public string ConvertToHtml()
        {
            using (TextReader unicodeFileStream = File.OpenText(_fullFilenameWithPath))
            {
                string html = string.Empty;

                string line = unicodeFileStream.ReadLine();
                while (line != null)
                {
                    html += HttpUtility.HtmlEncode(line);
                    html += "<br />";
                    line = unicodeFileStream.ReadLine();
                }

                return html;
            }
        }
    }
}

File: swift/HtmlConverter/Sources/UnicodeFileToHtmlTextConverter.swift
import Foundation

public class UnicodeFileToHtmlTextConverter {
    private let _fullFilenameWithPath: String

    public var fileName: String { _fullFilenameWithPath }

    public init(_ fullFilenameWithPath: String) {
        self._fullFilenameWithPath = fullFilenameWithPath
    }

    func convertToHtml() throws -> String {
        let lines = try String(contentsOfFile: fileName, encoding: .utf8)
            .components(separatedBy: .newlines)

        var html = ""
        for line in lines {
            html += HttpUtility.htmlEncode(line.trimmingCharacters(in: .whitespaces))
            html += "<br />"
        }
        html += "</html>"

        return html
    }
}
